### PR TITLE
rinetd: init at 0.73

### PR DESCRIPTION
**Pull request by Janik-Haag**
###### Description of changes
Trying to package rinetd but build falis because it trys to install stuff at /usr/local/sbin which off course doesn't work. When grepping through the sources it's only mentioned in the man page and the index.html which both don't theme like they set that. I also build the configure script and ran autoconfig and make but they also don't create files with a mention of users and make install just fails.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/rinetd/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  ];
+
+  configurePhase = ''
+    runHook preConfigure
+
+    ./bootstrap
+    ./configure
+
+    runHook postConfigure
+  '';
```
Comment (amjoseph-nixpkgs) [inline pkgs/servers/rinetd/default.nix:30]:
```suggestion
  preConfigure = ''
    ./bootstrap
  '';
```

File: pkgs/servers/rinetd/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoreconfHook
+
```
Comment (amjoseph-nixpkgs) [inline pkgs/servers/rinetd/default.nix:5]:
```suggestion
```

**Review by ghost — APPROVED** [review]
LGTM, with the change below to fix the build.

In general you shouldn't call `./configure` yourself; you should let `stdenv.mkDerivation` do that for you, because that way it does all sorts of important stuff like add the `configureFlags`.

**Comment by amjoseph-nixpkgs** [comment]
@ofborg build rinetd

**Comment by amjoseph-nixpkgs** [comment]
> rinetd on aarch64-darwin — Success
> rinetd on aarch64-linux — Success
> rinetd on x86_64-linux — Success

Nice!  I suggest you use your [shiny new commit-bit](https://github.com/NixOS/nixpkgs/issues/50105#issuecomment-1606082030) to merge this. :smile:

File: pkgs/servers/rinetd/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    repo = "rinetd";
+    rev = "v${version}";
+    hash = "sha256-W8PLGd3RwmBTh1kw3k8+ZfP6AzRhZORCkxZzQ9ZbPN4=";
+  };
+
+  nativeBuildInputs = [
+    autoreconfHook
+  ];
+
+  preConfigure= ''
```
Comment (RaitoBezarius) [inline pkgs/servers/rinetd/default.nix:22]:
```suggestion
  preConfigure = ''
```

File: pkgs/servers/rinetd/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    ./bootstrap
+  '';
+
+  meta = with lib; {
+    description = "TCP/UDP port redirector";
+    homepage = "https://github.com/samhocevar/rinetd";
+    changelog = "https://github.com/samhocevar/rinetd/blob/${src.rev}/CHANGES.md";
+    license = licenses.gpl2Only;
```
Comment (tilpner) [inline pkgs/servers/rinetd/default.nix:30]: is this correct? the readme says:

> Released under the terms of the GNU General Public License, version 2 or later.

Comment (Janik-Haag) [reply pkgs/servers/rinetd/default.nix]: Changed it should be fine now right?

**Comment by Janik-Haag** [comment]
@amjoseph-nixpkgs feel free to merge it, I already merged ~50 PRs in the last two or three days and one of my mentors basically made me promise to not do self merges if at all avoidable :D

**Comment by 0x4A6F** [comment]
Result of `nixpkgs-review pr 234520` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rinetd</li>
  </ul>
</details>

**Review by 0x4A6F — APPROVED** [review]

**Comment by 0x4A6F** [comment]
@amjoseph-nixpkgs could you not link to issues, that are watched by so many people.
And your suggestion to self merge could be a trap or a honest suggestion, I'm not sure what is worse.

@Janik-Haag you could have some sanity checks with `passthru.tests.version` utilising `testers.testVersion`.
```
Cannot nix-instantiate `rinetd.passthru.tests' because:
error: attribute 'tests' in selection path 'rinetd.passthru.tests' not found
```

**Comment by Janik-Haag** [comment]
@ofborg build

**Comment by 0x4A6F** [comment]
@ofborg  eval

**Comment by amjoseph-nixpkgs** [comment]
I think this wins the award for weirdest github comment I've seen so far:

> And your suggestion to self merge could be a trap or a honest suggestion, I'm not sure what is worse.

@0x4A6F it was of course an honest suggestion; how and why could it be "[a trap](https://knowyourmeme.com/memes/its-a-trap)"?  And how would an honest suggestion be worse?

**Comment by Janik-Haag** [comment]
@amjoseph-nixpkgs @0x4A6F could you please discuss this some where different like matrix to not ad notifications to this PR?

**Pull request merged**